### PR TITLE
feat: Add InputGroup component

### PR DESCRIPTION
**Pull request by GoOz**
Add new InputGroup component.

Basically it's a wrapper component with a prepend or append component that can be pretty much anything but it has been specially made for a `Text` component, an other `Input` or a `SelectBox`.

[Preview](https://gooz.github.io/cozy-ui/react/#!/InputGroup)

Fixes: #710

### CODE REVIEW

**Comment by GoOz** [comment]
[(𝐌) [UI] Ajouter le composant d'url](https://trello.com/c/eo92heRU/612-%F0%9D%90%8C-ui-ajouter-le-composant-durl)

**Comment by ptbrowne** [comment]
Field is a combination of Label and Input.
DualField seems more close to a Input.

I was thinking of renaming DualField to DualInput and then thought that it would be even simpler to have a side property in Input. If the implementation between the current DualField and Input are too different, Input could completely delegate to the current DualField implementation if it has a side property. What do you think ?

File: react/InputGroup/index.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      focused: false
+    }
+  }
+
+  handleFocus() {
+    this.setState(() => ({ focused: true }))
+  }
+
+  handleBlur() {
+    this.setState(() => ({ focused: false }))
```
Comment (ptbrowne) [inline react/InputGroup/index.jsx]:
```suggestion
    this.setState({ focused: false })
```

It is not necessary to use the functional variant of setState when state is not used to compute the new value of state.

File: react/InputGroup/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    super(props)
+    this.handleFocus = this.handleFocus.bind(this)
+    this.handleBlur = this.handleBlur.bind(this)
+    this.state = {
+      focused: false
+    }
+  }
+
+  handleFocus() {
+    this.setState(() => ({ focused: true }))
```
Comment (ptbrowne) [inline react/InputGroup/index.jsx]:
```suggestion
    this.setState({ focused: true })
```

File: stylus/components/forms.styl [hunk 3]
```diff
@@ -467,6 +467,49 @@ $field-inline
         flex-direction column
         margin-left 0
 
+$inputgroup
+    display inline-flex
+    flex-direction row
+    align-items stretch
+    width 100%
+    max-width rem(512)
```
Comment (ptbrowne) [inline stylus/components/forms.styl]: I am not sure we want to set a max-width here. It seems like an arbitrary limit.

File: react/InputGroup/Readme.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+### InputGroup with a appended text
```
Comment (drazik) [inline react/InputGroup/Readme.md]:
```suggestion
### InputGroup with an appended text
```

File: react/InputGroup/Readme.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+<form>
+  <div>
+    <InputGroup prepend={<Bold className="u-pl-1">text</Bold>}>
+      <Input placeholder="Placeholder" />
+    </InputGroup>
+  </div>
+</form>
+```
+
+### InputGroup with a appended input
```
Comment (drazik) [inline react/InputGroup/Readme.md]:
```suggestion
### InputGroup with an appended input
```

File: react/InputGroup/Readme.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+<form>
+  <div>
+    <InputGroup append={<Input placeholder="@domain.tld" className="u-w-4"/>}>
+      <Input placeholder="Email" />
+    </InputGroup>
+  </div>
+</form>
+```
+
+### InputGroup with a appended select
```
Comment (drazik) [inline react/InputGroup/Readme.md]:
```suggestion
### InputGroup with an appended select
```

File: react/InputGroup/Readme.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+<form>
+  <div>
+    <InputGroup append={<SelectBox inset options={options} className="u-w-4" />}>
+      <Input placeholder="URL" />
+    </InputGroup>
+  </div>
+</form>
+```
+
+### Full width InputGroup with a appended text
```
Comment (drazik) [inline react/InputGroup/Readme.md]:
```suggestion
### Full width InputGroup with an appended text
```

File: react/InputGroup/Readme.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+<form>
+  <div>
+    <InputGroup fullwidth append={<Bold className="u-pr-1">text</Bold>}>
+      <Input placeholder="Placeholder" />
+    </InputGroup>
+  </div>
+</form>
+```
+
+### Errored InputGroup with a appended text
```
Comment (drazik) [inline react/InputGroup/Readme.md]:
```suggestion
### Errored InputGroup with an appended text
```

**Comment by GoOz** [comment]
@drazik Very different, like… a lot. It looks the same but it's definitely not done the same way.

**Comment by drazik** [comment]
OK I would be in favor of branching directly in the `Input` component do avoid adding a new component. But I just saw the discussion on mattermost about this topic. Since the decision has already been discussed, that's ok

File: stylus/components/forms.styl (hunk 3 above)
Comment (GoOz) [reply stylus/components/forms.styl]: This is the same `max-width` the `input` has.

Comment (ptbrowne) [reply stylus/components/forms.styl]: A shared variable between input and inputgroup would be more explicit IMHO :)

**Pull request merged**